src/actions/profile-view.js

```diff
@@ -552,6 +552,42 @@ export function changeGlobalTrackOrder(globalTrackOrder: TrackIndex[]): Action {
   };
 }
 
+export function hideAllTracksByType(trackType: string): ThunkAction<void> {
+  return (dispatch, getState) => {
+    const globalTracks = getGlobalTracks(getState());
+    const globalIdsToHide = [];
+    const localIdsToHide = new Map();
+    for (const [index, track] of globalTracks.entries()) {
+      if (track.type === trackType) {
+        globalIdsToHide.push(index);
+      }
+      if (track.pid) {
+        const localTracksForPid = getLocalTracks(getState(), track.pid);
+
+        const idsToHideForPid = [];
+        for (const [localIndex, localTrack] of localTracksForPid.entries()) {
+          if (localTrack.type === trackType) {
+            idsToHideForPid.push(localIndex);
+          }
+        }
+        localIdsToHide.set(track.pid, idsToHideForPid);
+      }
+    }
```

Comment:
We need to look whether we're hiding too many tracks. We don't want to carry the action if If you look at `hideGlobalTrack`, you'll see that we don't carry on the action if we'll remove the last track. So you'll need to do something similar here too.

Reply:
Makes sense! I'll look into it!